packages/kbn-gen-ai-functional-testing/README.md

```diff
@@ -0,0 +1,49 @@
+# @kbn/gen-ai-functional-testing
+
+Package exposing various utilities for GenAI/LLM related functional testing.
+
+## Features
+
+### LLM connectors
+
+Utilizing LLM connectors on FTR tests can be done via the `getPreconfiguredConnectorConfig` and `getAvailableConnectors` tools.
+
+`getPreconfiguredConnectorConfig` should be used to define the list of connectors when creating the FTR test's configuration.
+
+```ts
+import { getPreconfiguredConnectorConfig } from '@kbn/gen-ai-functional-testing'
+
+export default async function ({ readConfigFile }: FtrConfigProviderContext) {
+  const xpackFunctionalConfig = {...};
+  const preconfiguredConnectors = getPreconfiguredConnectorConfig();
+
+  return {
+    ...xpackFunctionalConfig.getAll(),
+    kbnTestServer: {
+      ...xpackFunctionalConfig.get('kbnTestServer'),
+      serverArgs: [
+        ...xpackFunctionalConfig.get('kbnTestServer.serverArgs'),
+        `--xpack.actions.preconfigured=${JSON.stringify(preconfiguredConnectors)}`,
```

Comment:
Since we don't provision FTR config into the Cloud deployments, will these tests still be possible running against Cloud?
Is there a plan to extend it for serverless?

Reply:
> Since we don't provision FTR config into the Cloud deployments, will these tests still be possible running against Cloud?

Unfortunately I don't see how, given the tests relies on secrets that need to be provided, which Cloud will not. 

> Is there a plan to extend it for serverless?

There's no plan for it atm, given that this first test is mostly smoke testing LLM connector integration, which doesn't really differ between trad/serverless, but if we want to, I don't think anything would be blocking it. We would still need to figure out how to expose/retrieve the connector config for the serverless pipelines